engine/apps/api/views/alert_receive_channel.py

```diff
@@ -271,6 +273,25 @@ def send_demo_alert(self, request, pk):
 
         return Response(status=status.HTTP_200_OK)
 
+    @extend_schema(request=AlertReceiveChannelSerializer)
+    @action(detail=False, methods=["post"])
+    def test_connection(self, request):
+        # create in-memory instance to test with the (possible) unsaved data
+        data = request.data
+        # clear name while testing connection (to avoid name already used validation error)
+        data["verbal_name"] = None
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        instance = AlertReceiveChannel(**serializer.validated_data)
+        test_connection_func = getattr(instance.config, "test_connection", None)
+        if test_connection_func:
+            try:
+                test_connection_func(instance)
```

Comment:
maybe make `test_connection` return a boolean so exceptions are handled in `test_connection` itself? IMHO it would be better if internal API doesn't have to know about those implementation-specific details